File: cs1210/2019summer/assignments/hw15_clock-class/ClockTester.py
import Clock1
import Clock2
clockModule = Clock2

errorCount = 0

# Test Constructors
c1 = clockModule.Clock()
if c1.getTime() != (0,0,0):
    print('Error in constructor test 1')
    errorCount += 1

c2 = clockModule.Clock(10)
if c2.getTime() != (10,0,0):
    print('Error in constructor test 2')
    errorCount += 1
    
c3 = clockModule.Clock(10,15)
if c3.getTime() != (10,15,0):
    print('Error in constructor test 3')
    errorCount += 1
    
c4 = clockModule.Clock(10,15,35)
if c4.getTime() != (10,15,35):
    print('Error in constructor test 4')
    errorCount += 1
    
c5 = clockModule.Clock(minute=22,hour=14,second=44)
if c5.getTime() != (14,22,44):
    print('Error in constructor test 5')
    errorCount += 1

# Test Mutators
c6 = clockModule.Clock()
if c6.getTime() != (0,0,0):
    print('Error in mutator test 1')
    errorCount += 1

c6.setHour(10)
if c6.getTime() != (10,0,0):
    print('Error in mutator test 2')
    errorCount += 1

c6.setMinute(15)
if c6.getTime() != (10,15,0):
    print('Error in mutator test 3')
    errorCount += 1

c6.setSecond(35)
if c6.getTime() != (10,15,35):
    print('Error in mutator test 4')
    errorCount += 1
    
c7 = clockModule.Clock()
if c7.getTime() != (0,0,0):
    print('Error in mutator test 1')
    errorCount += 1

c7.setTime(13)
if c7.getTime() != (13,0,0):
    print('Error in mutator test 2')
    errorCount += 1
    
c7.setTime(23,24)
if c7.getTime() != (23,24,0):
    print('Error in mutator test 3')
    errorCount += 1
    

c7.setTime(33,34,35)
if c7.getTime() != (33,34,35):
    print('Error in mutator test 4')
    errorCount += 1
    

# Using named parameters
c7.setTime(hour=9)
if c7.getTime() != (9,34,35):
    print('Error in mutator test 5')
    errorCount += 1
    

c7.setTime(minute=10)
if c7.getTime() != (9,10,35):
    print('Error in mutator test 6')
    errorCount += 1
    
c7.setTime(second=11)
if c7.getTime() != (9,10,11):
    print('Error in mutator test 7')
    errorCount += 1
    

# With no parameters, the time is not changed.
c7.setTime()
if c7.getTime() != (9,10,11):
    print('Error in mutator test 8')
    errorCount += 1
    
# Reset
c7.reset()
if c7.getTime() != (0,0,0):
    print('Error in mutator test 9')
    errorCount += 1
    

#Test Accessors
c8 = clockModule.Clock(11,12,13)
if c8.getTime() != (11,12,13):
    print('Error in accessor test 1')
    errorCount += 1
    
if c8.getHour() != 11:
    print('Error in accessor test 2')
    errorCount += 1
    
if c8.getMinute() != 12:
    print('Error in accessor test 3')
    errorCount += 1
    
if c8.getSecond() != 13:
    print('Error in accessor test 4')
    errorCount += 1
    

if c8.getTime() != (11,12,13):
    print('Error in accessor test 5')
    errorCount += 1


# Test str()
c9 = clockModule.Clock(1,23,59)
if str(c9) != '01:23:59':
    print('Error in str() test 1')
    errorCount += 1
    
if errorCount == 0:
    print('No Errors Found')

File: csci295/2019fall/assignments/hw11_clock-interface/ClockTester.java


import java.util.ArrayList;



public class ClockTester {
    static ArrayList<TestResult> tests;

    private static String expected, result;
    
    public static void main(String[] args) {
        
        // Test Clock_1
        System.out.println("=================================================");
        System.out.println("Testing Clock_1");
        tests = new ArrayList<>();        
        clock_1ConstructorTests();
        runTests(new Clock_1());
        TestResult.reportTestResults(tests);
        
        // Test Clock_2
        System.out.println("=================================================");
        System.out.println("Testing Clock_2");
        tests = new ArrayList<>();        
        clock_2ConstructorTests();
        runTests(new Clock_2());
        TestResult.reportTestResults(tests);
    }

    public static void runTests(Clock c){
        c.setTime(1, 2, 3);
        expected = "1:2:3";
        result = c.toString();
        tests.add(new TestResult("Test 1", expected, result));
        
        c.setHour(11);
        expected = "11:2:3";
        result = c.toString();
        tests.add(new TestResult("Test 2", expected, result));

        c.setMinute(12);
        expected = "11:12:3";
        result = c.toString();
        tests.add(new TestResult("Test 3", expected, result));

        c.setSecond(13);
        expected = "11:12:13";
        result = c.toString();
        tests.add(new TestResult("Test 4", expected, result));
        
        c.reset();
        expected = "0:0:0";
        result = c.toString();
        tests.add(new TestResult("Test 5", expected, result));
        
        c.setTime(21, 22, 23);
        expected = "21:22:23";
        result = c.toString();
        tests.add(new TestResult("Test 6", expected, result));
        
        expected = "" + 21;
        result = "" + c.getHour();
        tests.add(new TestResult("Test 7", expected, result));

        expected = "" + 22;
        result = "" + c.getMinute();
        tests.add(new TestResult("Test 8", expected, result));

        expected = "" + 23;
        result = "" + c.getSecond();
        tests.add(new TestResult("Test 9", expected, result));

    }
    
    public static void clock_1ConstructorTests(){
        Clock c;
        
        c = new Clock_1();
        expected = "0:0:0";
        result = c.toString();
        tests.add(new TestResult("Constructor 1", expected, result));
        
        c = new Clock_1(10);
        expected = "10:0:0";
        result = c.toString();
        tests.add(new TestResult("Constructor 2", expected, result));

        c = new Clock_1(10, 11);
        expected = "10:11:0";
        result = c.toString();
        tests.add(new TestResult("Constructor 3", expected, result));

        c = new Clock_1(10, 11, 12);
        expected = "10:11:12";
        result = c.toString();
        tests.add(new TestResult("Constructor 4", expected, result));
    }

    public static void clock_2ConstructorTests(){
        Clock c;
        
        c = new Clock_2();
        expected = "0:0:0";
        result = c.toString();
        tests.add(new TestResult("Constructor 1", expected, result));
        
        c = new Clock_2(10);
        expected = "10:0:0";
        result = c.toString();
        tests.add(new TestResult("Constructor 2", expected, result));

        c = new Clock_2(10, 11);
        expected = "10:11:0";
        result = c.toString();
        tests.add(new TestResult("Constructor 3", expected, result));

        c = new Clock_2(10, 11, 12);
        expected = "10:11:12";
        result = c.toString();
        tests.add(new TestResult("Constructor 4", expected, result));
    }
    
}

/* Holds results of a single result */
class TestResult {

    private final String message;
    private final String expected;
    private final String received;

    public TestResult(String message, String expected, String received) {
        this.message = message;
        this.expected = expected;
        this.received = received;
    }
    
    private boolean passed() {
        return expected.equals(received);
    }
        
    public static void reportTestResults(ArrayList<TestResult> testResults) {
        System.out.println("Results of " + testResults.size() + " tests:");
        boolean errorsFound = false;
        for (TestResult result : testResults) {
            if (!result.passed()) {
                System.out.println("Error: " + result.message);
                System.out.println("\tExpected: \"" + result.expected + "\"");
                System.out.println("\tReceived: \"" + result.received + "\"");
                errorsFound = true;
            }
        }
        
        if(!errorsFound){
            System.out.println("No Errors Found.");
        }
    }    
} // End of class TestResult